packages/rocketchat-api/server/api.coffee

```diff
@@ -33,6 +33,14 @@ class API extends Restivus
 				success: false
 				error: msg or 'unauthorized'
 
+	getPaginationItems: (req) ->
+		console.log(req)
+		offset = if req.queryParams.offset then parseInt(req.queryParams.offset) else 0
+		count = if req.queryParams.count then parseInt(req.queryParams.count) else 25
+		count = 100 if count > 100
```

Comment:
If count is `-1`, then return count as `undefined` so it doesn't get applied and no special logic is required for everything..if the setting allows this ;)